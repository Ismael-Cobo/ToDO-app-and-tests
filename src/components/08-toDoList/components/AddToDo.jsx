import { useState } from "react"

export const AddToDo = ({handleAddToDo}) => {
    
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 2) return

        const newTodo = {
            id: new Date().getTime(),
            desc: inputValue,
            done: false
        };

        handleAddToDo(newTodo)

        setInputValue('')
    } 


    return (
        <form className='todo__form' onSubmit={handleSubmit}>
            <label className='form__title' htmlFor='todo'> Today I need to </label>
            <div className='form__input__group'>
                <input
                    type='text'
                    id='todo'
                    className='form__input'
                    value={inputValue}
                    onChange={handleInputChange}

                />
                <input
                    type='submit'
                    value='submit'
                    className='form__btn'
                />
            </div>
        </form>
    )
}
