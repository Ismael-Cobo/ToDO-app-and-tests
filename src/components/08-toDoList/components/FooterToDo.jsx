import React from 'react'

export const FooterToDo = () => {
    return (
        <footer className='todo__footer'>
            Inspired in <a href='https://codepen.io/lokifullap/pen/gOXNGBo' target={'_blank'} rel="noreferrer">Codepen</a>
        </footer>
    )
}
