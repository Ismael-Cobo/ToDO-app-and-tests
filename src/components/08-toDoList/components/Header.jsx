import LaptopSVG from '../../../styles/img/laptop.svg'

export const Header = () => {
    return (
        <div className='todo__heading'>
            <img className='heading__img' src={LaptopSVG} alt='heading Img' />
            <h1 className='heading__title'>To-Do List</h1>
        </div>
    )
}
