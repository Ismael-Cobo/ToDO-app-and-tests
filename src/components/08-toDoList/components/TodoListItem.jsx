import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleToggle }) => {
    return (
        <>
            <li
                className={`todo__item ${todo.done ? 'todo__done' : ''}`}
            >
                <p
                    className='todo__item__text'
                    onClick={() => handleToggle(todo.id)}
                >
                    {todo.desc}
                </p>
            </li>
            <button
                className='todo__item__delete'
                onClick={() => handleDelete(todo.id)}
            >
                Borrar
            </button>
        </>
    )
}
