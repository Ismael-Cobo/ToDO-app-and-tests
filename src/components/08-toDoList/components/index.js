export { AddToDo } from "./AddToDo";
export { Header } from "./Header";
export { ListToDo } from "./ListToDo";
export { FooterToDo } from "./FooterToDo";