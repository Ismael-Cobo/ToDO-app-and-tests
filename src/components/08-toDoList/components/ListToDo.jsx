import { TodoListItem } from "./TodoListItem"

export const ListToDo = ({ todos, handleDelete, handleToggle }) => {

    return (
        <section className='todo__items'>
            <ul className='todo__items__list'>
                {
                    todos
                        ? todos.map(todo => (
                            <div
                                className='todo__item__container'
                                key={todo.id}
                            >
                                <TodoListItem
                                    todo={todo}
                                    handleDelete={handleDelete}
                                    handleToggle={handleToggle}
                                />
                            </div>
                        ))
                        : ''
                }
            </ul>
        </section>
    )
}