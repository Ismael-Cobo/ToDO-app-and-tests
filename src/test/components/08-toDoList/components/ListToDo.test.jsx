import { shallow } from "enzyme"
import { ListToDo } from "../../../../components/08-toDoList/components/ListToDo"
import { demoReducer } from "../../../fixtures/demoTodos"

describe('test del componente <ListToDo />', () => {

    const todos = demoReducer
    const handleToggle = jest.fn()
    const handleDelete = jest.fn()

    const wrapper = shallow(
        <ListToDo
            todos={todos}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
        />
    )


    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener 3 <ListItems /> con sus props', () => {
        const listItems = wrapper.find('TodoListItem')

        expect(listItems.length).toBe(todos.length)
        expect(listItems.at(0).prop('handleToggle')).toEqual(expect.any(Function))
        expect(listItems.at(0).prop('handleDelete')).toEqual(expect.any(Function))
        expect(listItems.at(0).prop('todo')).toEqual(todos[0])
    })
})