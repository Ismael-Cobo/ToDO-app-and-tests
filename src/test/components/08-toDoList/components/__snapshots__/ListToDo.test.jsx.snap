// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test del componente <ListToDo /> debe de mostrarse correctamente 1`] = `
<section
  className="todo__items"
>
  <ul
    className="todo__items__list"
  >
    <div
      className="todo__item__container"
      key="1"
    >
      <TodoListItem
        handleDelete={[MockFunction]}
        handleToggle={[MockFunction]}
        todo={
          Object {
            "desc": "hacer esto",
            "done": false,
            "id": 1,
          }
        }
      />
    </div>
    <div
      className="todo__item__container"
      key="2"
    >
      <TodoListItem
        handleDelete={[MockFunction]}
        handleToggle={[MockFunction]}
        todo={
          Object {
            "desc": "Hacer aquello otro",
            "done": true,
            "id": 2,
          }
        }
      />
    </div>
    <div
      className="todo__item__container"
      key="4"
    >
      <TodoListItem
        handleDelete={[MockFunction]}
        handleToggle={[MockFunction]}
        todo={
          Object {
            "desc": "Hacer aquello otro2",
            "done": true,
            "id": 4,
          }
        }
      />
    </div>
  </ul>
</section>
`;
